### PR TITLE
Mocks HTTP tests

### LINKED ISSUE #21: Mock HTTP tests
For now we have two tests, `TestGitIgnoreExist` and `TestListFiles`, making external HTTP requests to be fulfilled.

We should mock those!

### PR DESCRIPTION
**Pull request by cuducos**
Fix #21 – This is my very first PR in Go, so feedback is surely welcomed!

I was inspired by [Learn Go with Tests](https://quii.gitbook.io/learn-go-with-tests/)'s strategy of mocking with a test HTTP server, which requires the URL to always be a parameter (that is to say, to avoid embedding the hard coded URL in the function bodies).

Not sure if fits though : )

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -21,24 +21,29 @@ const (
 )
 
 // return a slice of strings containing all the .gitignore file names from GitHub
-func listFiles() (files []string) {
+func listFiles(u string) (files []string) {
```
Comment (jonatasbaldin) [inline main.go]: Should we rename the `u` parameter to `url`? It's a bit more descriptive.

File: main.go [hunk 2]
```diff
@@ -21,24 +21,29 @@ const (
...
 			unscapedHref, _ := url.QueryUnescape(href)
-			files = append(files, (path.Base(strings.TrimSuffix(unscapedHref, GitIgnoreExt))))
+			files = append(files, path.Base(strings.TrimSuffix(unscapedHref, GitIgnoreExt)))
 		}
 	})
-	c.Visit(GHGitignoreRepo)
+
+	err := c.Visit(u)
+	if err != nil {
+		fmt.Println(u)
```
Comment (jonatasbaldin) [inline main.go]: We can merge this two "log" lines like this:
```go
log.Fatalf("couldn't visit url %q, err: %q", url, err)
```

`%q` will add double quotes in those two parameters.

Comment (cuducos) [reply main.go]: Yay! Thanks for the tip : )

File: main.go (hunk 1 above)
Comment (cuducos) [reply main.go]: `url` will conflict with `net/url` package namespace (that's why I shortened it to `u`). I'll go with `pageUrl` then ; )

Comment (jonatasbaldin) [reply main.go]: Nice, thanks!

**Comment by jonatasbaldin** [comment]
Thanks @cuducos, merged it!

**Pull request merged**